cfme/storage/volume_backup.py

```diff
@@ -0,0 +1,249 @@
+        return False
+
+    volume_name = BootstrapSelect(name='volume_id')
+    save = Button('Save')
+    reset = Button('Reset')
+    cancel = Button('cancel')
+
+
+@attr.s
+class VolumeBackup(BaseEntity, WidgetasticTaggable):
+    """ Model of an Storage Volume Backups in cfme
+
+    Args:
+        name: name of the backup
+        provider: provider
+    """
+    name = attr.ib()
+    provider = attr.ib()
+
+    def restore(self, name):
+        """Restore the volume backup. this feature included in 5.9 and above.
+
+        Args:
+            name: volume name
+        """
+        view = navigate_to(self, 'restore')
+        view.volume_name.fill(name)
+        view.save.click()
+        view.flash.assert_success_message('Restoring Cloud Volume "{}"'.format(self.name))
+
+    def refresh(self):
+        self.provider.refresh_provider_relationships()
+        self.browser.refresh()
+
+    @property
+    def size(self):
+        """ size of cloud volume backup.
+
+        Returns:
+            :py:class:`int' size of volume backup.
+        """
+        view = navigate_to(self, 'Details')
+        return view.entities.properties.get_text_of('Size')
+
+    @property
+    def status(self):
+        """ Present status of cloud volume backup.
+
+        Returns:
+            :py:class:`str' status [available/restoring] of volume backup.
+        """
+        view = navigate_to(self, 'Details')
+        return view.entities.properties.get_text_of('Status')
+
+    @property
+    def volume(self):
+        """ volume name of backup.
+
+        Returns:
+            :py:class:`str' respective volume name.
+        """
+        view = navigate_to(self, 'Details')
+        return view.entities.relationships.get_text_of('Cloud Volume')
+
+
+@attr.s
+class VolumeBackupCollection(BaseCollection):
+    """Collection object for :py:class:'cfme.storage.volume_backups.VolumeBackup' """
+    ENTITY = VolumeBackup
+
+    def all(self):
+        """returning all backup objects"""
+        view = navigate_to(self, 'All')
+        backups = [self.instantiate(name=item, provider=self.filters.get('provider'))
+                   for item in view.entities.all_entity_names]
+        return backups
+
+    def delete(self, *backups):
+        """Delete one or more backups
+
+        Args:
+            One or Multiple 'cfme.storage.volume_backup.VolumeBackup' objects
+        """
+
+        view = navigate_to(self, 'All')
+
+        if view.entities.get_all():
+            for backup in backups:
+                try:
+                    view.entities.get_entity(backup.name).check()
+                except ItemNotFound:
+                    raise BackupNotFound("Volume backup {} not found".format(backup.name))
+
+            view.toolbar.configuration.item_select('Delete selected Backups', handle_alert=True)
+
+            for backup in backups:
+                wait_for(
+                    lambda: backup.name not in view.entities.all_entity_names,
```

Comment:
it makes sense to check all names together instead of one by one.

Reply:
@izapolsy I think still we need compress list using list compression as we are looking for back.name. Please, can you any other idea for comparison?

Reply:
I guess you can use something like:
bool(set([backup.name for backup in backups]) & set(view.entities.all_entity_names))